delegator.py

```diff
@@ -182,7 +182,11 @@ def run(self, block=True, binary=False, cwd=None, env=None):
                 popen_kwargs['cwd'] = cwd
             if env:
                 popen_kwargs['env'].update(env)
-            s = subprocess.Popen(self._popen_args, **popen_kwargs)
+            args = self._popen_args
+            if not isinstance(args, str):
+                # It is recommended to pass a sequence based on docs.
+                args = subprocess.list2cmdline(args)
```

Comment:
`subprocess.list2cmdline` is used mostly (only?) on Windows, so using it here might have unexpected results on *nix systems.

Reply:
An alternative could be to use `shlex` here, but required routines are only Python3.3+?

Reply:
Maybe it's better to set `shell` kwargs depending on what the given `args` are.
Using `shlex` and/or `subprocess.list2cmdline` doesn't work in all cases, I guess.

More over `shell=True` should only be used if the cmd line can't be represented with a sequence.